### ISSUE TITLE
StringStyle.swift cause crash

### ISSUE THREAD

**Issue by wenjingjie** [Rightpoint/BonMot#424]
hi, our project got this crash log in firebase when we are using   pod 'BonMot', '6.0.0'. seems related to the StringStyle.swift. 
please take a look. thanks a lot.

Crashed: com.apple.main-thread
EXC_BAD_ACCESS KERN_PROTECTION_FAILURE 0x00000001ef601f60
Crashed: com.apple.main-thread
0  ???                            0x1ef601f60 (Missing)
1  HLLDriver-SEA                  0xdb6e8 StringStyle.attributes.getter + 122 (StringStyle.swift:122)
2  HLLDriver-SEA                  0xe45984 LMDFontStyle.attributes.getter + 230 (Fonts.swift:230)
3  HLLDriver-SEA                  0xaee304 closure #1 in static UITabBarItem.homeTabBarItem(image:title:) + 33 (UIView+Home.swift:33)
4  HLLDriver-SEA                  0xa48b20 closure #1 in HLLTabBarController.setupTabs() + 31 (UIView+Home.swift:31)
5  HLLDriver-SEA                  0xa47318 HLLTabBarController.setupTabs() + 2838604 (<compiler-generated>:2838604)
6  HLLDriver-SEA                  0xa47be8 @objc HLLTabBarController.viewDidLoad() + 34 (HLLTabBarController.swift:34)
7  UIKitCore                      0x37c7b4 -[UITabBarController initWithNibName:bundle:] + 196
8  HLLDriver-SEA                  0xa48164 @objc HLLTabBarController.init(nibName:bundle:) + 2842264 (<compiler-generated>:2842264)
9  HLLDriver-SEA                  0xadd650 RootFlowController.handleFlowEvent(_:) + 93 (RootFlowController.swift:93)
10 HLLDriver-SEA                  0xaddce8 @objc RootFlowController.handleFlowEvent(_:) + 3455516 (<compiler-generated>:3455516)
11 HLLDriver-SEA                  0xf23084 @objc UIResponder.handleFlowEvent(_:) + 1240980 (<compiler-generated>:1240980)
12 HLLDriver-SEA                  0xf23084 @objc UIResponder.handleFlowEvent(_:) + 1240980 (<compiler-generated>:1240980)
13 HLLDriver-SEA                  0xf23084 @objc UIResponder.handleFlowEvent(_:) + 1240980 (<compiler-generated>:1240980)
14 HLLDriver-SEA                  0xf23084 @objc UIResponder.handleFlowEvent(_:) + 1240980 (<compiler-generated>:1240980)
15 HLLDriver-SEA                  0xf23084 @objc UIResponder.handleFlowEvent(_:) + 1240980 (<compiler-generated>:1240980)
16 HLLDriver-SEA                  0x9ef3a4 __35-[SplashScreenVC goLoginOrMainPage]_block_invoke.420 + 158 (SplashScreenVC.m:158)
17 UIKitCore                      0x1024410 -[UIViewAnimationBlockDelegate _didEndBlockAnimation:finished:context:] + 688
18 UIKitCore                      0xff8ba8 -[UIViewAnimationState sendDelegateAnimationDidStop:finished:] + 244
19 UIKitCore                      0xff90d8 -[UIViewAnimationState animationDidStop:finished:] + 240
20 QuartzCore                     0x171550 CA::Layer::run_animation_callbacks(void*) + 276
21 libdispatch.dylib              0x61298 _dispatch_client_callout + 16
22 libdispatch.dylib              0xfce0 _dispatch_main_queue_callback_4CF$VARIANT$mp + 904
23 CoreFoundation                 0x9a998 __CFRUNLOOP_IS_SERVICING_THE_MAIN_DISPATCH_QUEUE__ + 12
24 CoreFoundation                 0x94df8 __CFRunLoopRun + 2528
25 CoreFoundation                 0x93ed0 CFRunLoopRunSpecific + 572
26 GraphicsServices               0x3570 GSEventRunModal + 160
27 UIKitCore                      0xb302d0 -[UIApplication _run] + 1052
28 UIKitCore                      0xb3584c UIApplicationMain + 164
29 HLLDriver-SEA                  0x48cc main + 36 (main.m:36)
30 libdyld.dylib                  0x1140 start + 4

**Comment by ZevEisenberg** [comment]
Is this crash reproducible? If so, do you have a sample project that reproduces it? Or did you get this as a crash log in the wild? I'm also curious about stack frame 0, which seems to have missing symbols.

**Comment by wenjingjie** [comment]
thanks for reply, it can not be reproducible, just the crash log from firebase log, the stack frame 0 is missing

**Comment by wenjingjie** [comment]
> Is this crash reproducible? If so, do you have a sample project that reproduces it? Or did you get this as a crash log in the wild? I'm also curious about stack frame 0, which seems to have missing symbols.


thanks for reply, it can not be reproducible, just the crash log from firebase log, the stack frame 0 is missing

**Comment by ZevEisenberg** [comment]
I had a look at the code at that spot and didn't see anything suspicious. Have you tried updating to BonMot 6.1.1? Also, have you received many reports of this crash? Or just one?

**Comment by wenjingjie** [comment]
> I had a look at the code at that spot and didn't see anything suspicious. Have you tried updating to BonMot 6.1.1? Also, have you received many reports of this crash? Or just one?

yes, just one crash. I will update to BonMot 6.1.1 and keep looking, thanks~~~

**Comment by wenjingjie** [comment]
> 

hi, we got another crash report. the log is below:

Crashed: com.apple.main-thread
0  libswiftCore.dylib             0x2ee4d4 swift_release + 36
1  HLLDriver-SEA                  0xe68e4 destroy for StringStyle + 4366674148 (<compiler-generated>:4366674148)
2  HLLDriver-SEA                  0xef1c8 outlined destroy of StringStyle + 4366709192 (<compiler-generated>:4366709192)
3  libobjc.A.dylib                0x77d0 object_cxxDestructFromClass(objc_object*, objc_class*) + 112
4  libobjc.A.dylib                0x4858 objc_destructInstance + 88
5  libobjc.A.dylib                0xd7d4 _objc_rootDealloc + 52
6  libobjc.A.dylib                0x26990 _object_remove_associations + 628
7  libobjc.A.dylib                0x4868 objc_destructInstance + 104
8  libobjc.A.dylib                0xd7d4 _objc_rootDealloc + 52
9  UIKitCore                      0x2e3af0 <redacted> + 152
10 UIKitCore                      0x1e3ae4 <redacted> + 936
11 UIKitCore                      0x4377d0 <redacted> + 180
12 libsystem_blocks.dylib         0x1840 _Block_release + 184
13 UIKitCore                      0x1dacb0 <redacted> + 24
14 libsystem_blocks.dylib         0x1840 _Block_release + 184
15 CoreFoundation                 0xc350 <redacted> + 216
16 UIKitCore                      0x2aa7e4 <redacted> + 240
17 UIKitCore                      0x17f200 <redacted> + 244
18 UIKitCore                      0x192bc8 <redacted> + 240
19 QuartzCore                     0xc09c4 <redacted> + 276
20 libdispatch.dylib              0x64094 <redacted> + 16
21 libdispatch.dylib              0x10d44 <redacted> + 928
22 libdispatch.dylib              0x10994 <redacted> + 36
23 CoreFoundation                 0x4e0d4 <redacted> + 12
24 CoreFoundation                 0xb5f8 <redacted> + 2544
25 CoreFoundation                 0x1e250 CFRunLoopRunSpecific + 572
26 GraphicsServices               0x1988 GSEventRunModal + 160
27 UIKitCore                      0x4e5df8 <redacted> + 1080
28 UIKitCore                      0x27f490 UIApplicationMain + 336
29 HLLDriver-SEA                  0x48cc main + 36 (main.m:36)
30 ???                            0x1047844d0 (Missing)
